test/rules/assertions/requireParamName.js

```diff
@@ -31,6 +31,28 @@ export default {
           message: 'There must be an identifier after @param tag.'
         }
       ]
+    },
+    {
+      code: `
+          /**
+           * @param foo
+           */
+          function quux (foo) {
+
+          }
+      `,
+      errors: [
+        {
+          message: 'Unexpected tag `@param`'
```

Comment:
This error is already reported by `check-tag-names`, seems like a single responsibility principle breakage for me. It also make `settings.preferredTagName` a dependency for these dozens of rules.

Reply:
I don't see that as a problem in this case. The preexisting code was checking this I think so it would avoid the need to check for all of the possible aliases, and eslint settings are intended to be shared across rules for such use cases as this. That test is to confirm that the behavior which occurs in `check-tag-names` (which reports an error message more precise to the rule's purpose as opposed to this example which is just reporting an unusual scenario where the user has blocked a parameter the rule depends on) also occurs in the rules which rely on it.

I suggest raising a separate issue if you still wish to remove the dependency as this is a preexisting behavior (though one which I think makes sense--otherwise, we'd have to check all possible aliases or force users to repeat their preferences for each rule).

Reply:
That make sense then.